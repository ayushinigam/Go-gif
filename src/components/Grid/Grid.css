.gridContainer {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    margin-top: 30px;
}