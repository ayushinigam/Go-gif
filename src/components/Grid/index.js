import React from 'react';
import PropTypes from 'prop-types'; 
import Gif from '../Gif';
import './Grid.css';

class Grid extends React.Component {
    gridItems = () => {
        const {gridData} = this.props;
        return gridData.map((gridItemData) =>  <Gif imageData={gridItemData} key={gridItemData.id}/>)
    }
    render() {
        
        return (
            <div className='gridContainer'>
                { this.gridItems()}
            </div>
        )
    }
}

Grid.defaultProps = {
    gridData: []
}

Grid.propTypes = {
    gridData: PropTypes.array
}

export default Grid