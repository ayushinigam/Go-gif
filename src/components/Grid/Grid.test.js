import React from 'react';
import renderer from 'react-test-renderer';
import Grid from './index';

describe(('App'), () => {
    const gridData = [{
        moving: {height: 10, width: 10, url: 'test/moving'},
        still: {height: 10, width: 10, url: 'test/still'},
        id: '1'
    }]
it('renders correctly', () => {
  const tree = renderer.create(<Grid gridData={gridData}/>).toJSON();
  expect(tree).toMatchSnapshot();
});
});
