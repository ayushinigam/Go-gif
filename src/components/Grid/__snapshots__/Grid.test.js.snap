// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div
  className="gridContainer"
>
  <div
    className="imageContainer"
    onClick={[Function]}
  >
    <img
      className="imageControl"
      src={
        Object {
          "ReactComponent": Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          },
          "__esModule": true,
          "default": "play.svg",
        }
      }
    />
    <img
      className="gifImage"
      height={10}
      src="test/still"
      style={
        Object {
          "backgroundColor": "pink",
        }
      }
      width={10}
    />
  </div>
</div>
`;
